### DISCUSSION TITLE
Login not possible, Contact Administrator

### DISCUSSION THREAD

**Discussion by sanjaypaudel35** [zitadel/zitadel#7178] [Q&A] [answered]
when i login from the user created using the zitadel https://id.hdlcommerce.cloud/v2beta/users/human api it says following error
after login.
Login not possible. The user is required to have at least one grant on the application. Please contact your administrator. (Internal)

**Comment by hifabienne** [comment, marked as answer]
hei @sanjaypaudel35 
This indicates, that you have enabled the check authorization on authentication on your project, where the user tries to login.
<img width="697" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/03927ece-6754-4946-aa55-e00162051e77">

If this is enabled, the authenticating user needs an authorization on that project. you can either disable if it is not intended, or you can give the user a authorization/usergrant

**Reply by Bratn** [reply]
It seems like this error message isn't customizable. Is this intended or a bug?

**Reply by hifabienne** [reply]
at the moment this is intended, error messages are not customizable

**Reply by nrvnrvn** [reply]
@hifabienne what if there are two users, one of them has a grant, another one does not. Imagine a clean session: end user tries to login to an app integrated with Zitadel using the 2nd user, ends up with this error screen. But then end user goes to their app's login page and clicks "login via Zitadel" intending to use the 1st user's credentials. Then they are immediately taken to the same error screen without the ability to switch the user or enter different credentials.

Is there a way to "escape this login trap"?

**Reply by hifabienne** [reply]
so does that mean those two users belong to the same user? i have not tested this exact use case, but it feels like a but, in that case please open up a bug report

**Reply by nrvnrvn** [reply]
two users belong to the same org, and one is granted access to project and another one is not.

I am looking at this from the UX perspective primarily. The valid option is to go to /ui/console and sign yourself out so that you can authenticate with another pair of credentials. And as a user I'd like to have a link elsewhere from the error page(to my profile or just a sign out button probably). But looking at how others have addressed this I see the same behaviour with Microsoft(see https://learn.microsoft.com/en-us/troubleshoot/entra/entra-id/app-integration/error-code-aadsts50105-user-not-assigned-role). 

If we assume that the whole app-IDP system operates in a single user browser environment(so to speak), this is fine just to have a concise message that guides you to talk to your admin.

If the same person needs to handle multiple credentials and switch between users if the first user is the "wrong" one they need to know where to go to sign out and the error screen does not provide any links which _might_ be confusing for some users.
